dataherald/db_scanner/sqlalchemy.py

```diff
@@ -89,6 +89,27 @@ def get_processed_column(
                 data_type=str(column["type"]),
                 low_cardinality=False,
             )
+
+        # special case for PostgreSQL table - read query planner statistics from the pg_stats view
+        # TODO doesn't work for views, only tables
+        if db_engine.engine.driver == "psycopg2":
+            # TODO escape table and column names
+            rs = db_engine.engine.execute(
+                f"SELECT n_distinct, most_common_vals::TEXT::TEXT[] FROM pg_catalog.pg_stats WHERE tablename = '{table}' AND attname = '{column['name']}'"
+            ).fetchall()
+            if MIN_CATEGORY_VALUE < rs[0]["n_distinct"] <= MAX_CATEGORY_VALUE:
+                return ColumnDetail(
+                    name=column["name"],
+                    data_type=str(column["type"]),
+                    low_cardinality=True,
+                    categories=rs[0]["most_common_vals"],
```

Comment:
I apologize for the delay in my response. I've been reviewing the code, and it appears that the `rs[0]["most_common_vals"]` field, as its name suggests, only retrieves a limited number of values, rather than all of them. In our current context, we require access to all distinct values. I'm considering a solution where, given the low cardinality of this field, we can execute a new SQL query using either `GROUP BY` or `DISTINCT` to retrieve all the necessary values for populating the `categories`. 
What are your thoughts on this approach?

Reply:
Running a SQL query like that would do a full table scan if the column has no index, which is the thing I'm trying to avoid.

Reply:
According to the [PostgreSQL documentation](https://www.postgresql.org/docs/16/planner-stats.html#:~:text=The%20amount%20of,presently%20100%20entries.):
```
The amount of information stored in pg_statistic by ANALYZE, in particular the maximum number of entries in the most_common_vals and histogram_bounds arrays for each column, can be set on a column-by-column basis using the ALTER TABLE SET STATISTICS command, or globally by setting the [default_statistics_target](https://www.postgresql.org/docs/16/runtime-config-query.html#GUC-DEFAULT-STATISTICS-TARGET) configuration variable. The default limit is presently 100 entries.
```

Reply:
From a high level what we are doing here is using the PostgreSQL definition of "low cardinality" instead of what is hard-coded into DataHerald. I would argue that the database is in a better position to define what "low cardinality" should mean since it knows more about the data. PostgreSQL also allows the DBA to manually tune the definition of "low cardinality" as described in the documentation above, which is more flexible than the hard-coded value in DataHerald.

Reply:
it's ok. Thanks for your contribution